jaxdecomp/_src/transpose.py

```diff
@@ -150,7 +166,14 @@ def partition(kind: str, mesh: Mesh, arg_infos: Tuple[ShapeDtypeStruct],
     input_sharding = NamedSharding(mesh, P(*arg_infos[0].sharding.spec))
     output_sharding = NamedSharding(mesh, P(*result_infos.sharding.spec))
     global_shape = arg_infos[0].shape
-    pdims = (get_axis_size(input_sharding, 0), get_axis_size(input_sharding, 1))
```

Comment:
Make sure that `pdims`  provided to the c++ primitive is the original one (same reason as above)
But in the JAX abstract eval we need the reversed pdims because it changes the pencil shape
So c++ `pdims` are the original and cudecomp knows how to handle them
We mirror cudecomp reversal in our abstract eval